### DISCUSSION TITLE
Detect whether media is playing and count time active in tab if so

### DISCUSSION THREAD

**Discussion by CorvetteCole** [proginosko/LeechBlockNG#356] [Ideas]
Right now if I have "Count time spent on these sites only when browser tab is active" with YouTube, then playing a video then clicking away pauses the time countdown. I think if media is playing, the tab should also count as "active" so it takes from the countdown.

This would prevent inactive YouTube tabs that have no media from counting as time, but wouldn't allow somebody to bypass the counter easily.

**Comment by proginosko** [comment]
I haven't been able to reproduce this. Can you be more specific about what you mean by "clicking away"? Perhaps give a series of steps to reproduce the issue.

Also, which OS are you using?

**Reply by dawin64** [reply]
I have the same problem, it would be nice to have an option to only count down the time when media is playing. and to reproduce it, you can open a youtube video, put it on a different monitor and do something on the main monitor, the time does not count.

**Reply by proginosko** [reply]
It's possible for an extension to detect whether a tab is producing audio, which would presumably do the trick. In that case, it looks like users would want three options:

1. Count time _only_ when tab is producing audio
2. Count time _always_ when tab is producing audio (i.e., even if tab is not active)
3. Ignore whether tab is producing audio

Right?

**Reply by dawin64** [reply]
I think so, I don't know if this is possible with these solutions, but I think it should also be possible to count only the time when a tab is playing audio, regardless of whether the tab is active or not. (not count time when audio doesn't play)

**Comment by CorvetteCole** [comment]
This is on Linux with Wayland (Arch Linux specifically). I'll get some videos demonstrating this

**Comment by proginosko** [comment]
Added option "Count time spent on these sites only when browser tab is playing audio" in version 1.6.4.
